aws-lc-rs/src/ec/key_pair.rs

```diff
@@ -163,6 +190,49 @@ impl EcdsaKeyPair {
         }
     }
 
+    /// Deserialize a DER private key and produce an ECDSA key.
+    ///
+    /// This function will attempt to automatically detect the underlying key format, and
+    /// supports the unencrypted PKCS#8 `PrivateKeyInfo` structures as well as key type specific
+    /// formats.
+    ///
+    /// # Errors
+    /// `error::KeyRejected` if parsing failed or key otherwise unacceptable.
+    ///
+    /// # Panics
+    pub fn from_private_key_der(
+        alg: &'static EcdsaSigningAlgorithm,
+        private_key: &[u8],
+    ) -> Result<Self, KeyRejected> {
+        unsafe {
+            let mut out = std::ptr::null_mut();
+            if aws_lc::d2i_AutoPrivateKey(
```

Comment:
I think it would be better to invoke `d21_PrivateKey` with `EVP_PKEY_EC`. 

```
...
    if aws_lc::d2i_PrivateKey(
        EVP_PKEY_EC,
        &mut out,
        &mut private_key.as_ptr(),
...
```